### PR TITLE
EZP-24549: edit content in given language

### PR DESCRIPTION
**Pull request by mhyndle**
Jira: https://jira.ez.no/browse/EZP-24549

# Description
When viewing the content we are viewing it in given language. When clicking the "Edit" button placed on Action bar we should edit the content in currently selected language.

# Tasks
- [x] Implementation
- [x] Tests
- [x] Rebase

### CODE REVIEW

File: Resources/public/js/views/services/ez-contenteditviewservice.js [hunk 1]
```diff
@@ -116,14 +125,16 @@ YUI.add('ez-contenteditviewservice', function (Y) {
...
          */
-        _loadModel: function (attr, modelId, errorMsg, callback) {
-            var model = this.get(attr);
+        _loadModel: function (attr, modelId, options, errorMsg, callback) {
+            var model = this.get(attr),
+                loadOptions = {api: this.get('capi')};
 
             model.set('id', modelId);
-            model.load({api: this.get('capi')}, Y.bind(function (error) {
+            model.load(Y.merge(loadOptions,options), Y.bind(function (error) {
```
Comment (dpobel) [inline Resources/public/js/views/services/ez-contenteditviewservice.js]: nitpick: space after the `,`

File: Resources/public/js/views/services/ez-contenteditviewservice.js [hunk 2]
```diff
@@ -36,10 +36,12 @@ YUI.add('ez-contenteditviewservice', function (Y) {
          */
         _load: function (next) {
             var request = this.get('request'),
-                service = this;
+                service = this,
+                languageCode = request.params.languageCode;
 
+            this.set('languageCode', languageCode);
```
Comment (dpobel) [inline Resources/public/js/views/services/ez-contenteditviewservice.js]: this should not be done here, you should add an event handler on the `requestChange` event for that (and the corresponding test)

File: Resources/public/js/views/services/ez-locationviewviewservice.js [hunk 3]
```diff
@@ -240,12 +243,13 @@ YUI.add('ez-locationviewviewservice', function (Y) {
                 type = this.get('contentType'),
                 discoveryService = this.get('capi').getDiscoveryService();
 
+            this.set('languageCode', request.params.languageCode);
```
Comment (dpobel) [inline Resources/public/js/views/services/ez-locationviewviewservice.js]: same as https://github.com/ezsystems/PlatformUIBundle/pull/278/files#r34654568

**Comment by dpobel** [comment]
You should add a test (or an assertion ?) in the tests of the ContentEditViewService to check that the content.load method receives the expected language.

In addition, did you check that these changes do not break the ContentCreateViewService (it extends the ContentEditViewService) ?

File: Resources/public/js/views/services/ez-contenteditviewservice.js [hunk 4]
```diff
@@ -24,6 +24,7 @@ YUI.add('ez-contenteditviewservice', function (Y) {
     Y.eZ.ContentEditViewService = Y.Base.create('contentEditViewService', Y.eZ.ViewService, [], {
         initializer: function () {
             this.on('*:closeView', this._handleCloseView);
+            this.on('*:requestChange', this._handleRequestChange);
```
Comment (dpobel) [inline Resources/public/js/views/services/ez-contenteditviewservice.js]: I would rename `_handleRequestChange` to `_setLanguageCode` or something similar, that way you directly what happen when the request change.

File: Resources/public/js/views/services/ez-locationviewviewservice.js [hunk 5]
```diff
@@ -26,6 +26,7 @@ YUI.add('ez-locationviewviewservice', function (Y) {
             this.on('*:editAction', this._editContent);
             this.on('*:sendToTrashAction', this._sendContentToTrashConfirmBox);
             this.on('*:moveAction', this._selectLocation);
+            this.on('*:requestChange', this._handleRequestChange);
```
Comment (dpobel) [inline Resources/public/js/views/services/ez-locationviewviewservice.js]: same as above

**Comment by mhyndle** [comment]
About creating a content it seems that it's not broken. Content is being created with mainLanguage, for example: if you are displaying folder in french translation and main language is eng-GB, then creating content leads to create new content in mainLanguage not the one which parent content was being previewed in. I suppose that's correct behaviour.

File: Resources/public/js/views/services/ez-contenteditviewservice.js [hunk 6]
```diff
@@ -36,10 +39,11 @@ YUI.add('ez-contenteditviewservice', function (Y) {
          */
         _load: function (next) {
             var request = this.get('request'),
-                service = this;
+                service = this,
+                languageCode = request.params.languageCode;
```
Comment (dpobel) [inline Resources/public/js/views/services/ez-contenteditviewservice.js]: so you should use this.get('languageCode') not the request.

**Comment by mhyndle** [comment]
ready for review
ping @dpobel @yannickroger @StephaneDiot

**Comment by dpobel** [comment]
+1

@yannickroger @StephaneDiot please review

**Comment by StephaneDiot** [comment]
+1

**Pull request merged**